### ISSUE TITLE
JS error after clicking on edit button

### ISSUE THREAD

**Issue by alberskib** [darul75/ng-prettyjson#13]
Hey!

First of all, thanks for great lib!

When you open JS console and click on edit button toy will notice error:
`Uncaught ReferenceError: define is not defined` which comes from `mode-json.js`.
You are able to notice it even in demo. 
Just download repo, `bower install` and try out.

If you need more info how to reproduce the problem, please let me know.
Thanks!

**Comment by darul75** [comment]
yes indeed, it is relative to "ace" library, I did not check it well until now, you can see code here 

https://github.com/darul75/ng-prettyjson/blob/gh-pages/javascripts/ng-prettyjson.js#L78

```javascript
editor.getSession().setMode("ace/mode/json");  
```

Then I guess they use requirejs to load all stuff but I can not check it now.

Let me know if you find, ace solution must be there http://ace.c9.io/#nav=embedding

**Comment by alberskib** [comment]
There are two ways to solve this issue:
1. Add require.js to your project - you will not receive error but json will not be colored during edition
2. Replace file mode-json.js with next file [mode-json.js](https://github.com/ajaxorg/ace-builds/blob/master/src-noconflict/mode-json.js) (from non conflig directorory) - as a result you will not recieve any error and json will be colored during edition.

**Comment by darul75** [comment]
Thx @alberskib for providing such details, I will apply and document it when back home.

**Comment by pynej** [comment]
Also worth noting, you can include just the core `ace.js` in the src folder, and then comment out the following line in `ng-prettyjson.js` and you will get editing capabilities without needing to load the entire ace solution.  We do this and lose color hilighting and other functions, but have a working editable region.

`editor.getSession().setMode("ace/mode/json");`
